src/components/data-table/column-sorting/column-sort.service.ts

```diff
@@ -0,0 +1,43 @@
+import {Injectable} from '@angular/core';
+import {Subject} from 'rxjs/Subject';
+import {
+  SortDirection,
+  IColumnSortingModel
+} from './column-sort';
+
+/**
+ * Service handling column sort change events as an observable stream.
+ */
+@Injectable()
+export class MdDataColumnSortingService {
+  public sortingColumn$: Subject<IColumnSortingModel> = new Subject();
+
+  /**
+   * Programatically set the column / direction of sorting
+   * @param sorting - column & direction to sort by.
+   */
+  public setSorting(sorting: IColumnSortingModel) {
+    this.sortingColumn$.next(sorting);
+  }
+
+  /**
+   * Change sorting based on user action on on a particular column heading.
+   * Inputting the currently sorted-by column will invert the sort direction.
+   * @param id - identifier of column to sort by
+   * @param sorting - current sorting information
+   */
+  public changeSorting(fromId:string, sorting:IColumnSortingModel) {
+    let newSort: IColumnSortingModel = {
+      column: fromId,
+      direction: SortDirection.ASCEND
+    };
+
+    if (sorting && sorting.column === fromId) {
+      // invert currently selected column
+      newSort.direction = sorting.direction === SortDirection.DESCEND ?
+          SortDirection.ASCEND : SortDirection.DESCEND;
+    }
+
+    this.setSorting(newSort);
```

Comment:
As the doc comments indicate, these have two separate purposes. `setSorting` could be used to simply assign the default for the view, for instance, whereas `changeSorting` is really more of an event handler, so maybe it should be named that way. `onColumnChange` or `onSortChange` or something. **`onColumnSelect`**
(It's in the service because of style-guide comments to keep only 'display logic' in components, and it felt like controller/presenter logic to me.)